compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtBase.td

```diff
@@ -38,4 +39,31 @@ class RankedTensorOrMemRefOf<list<Type> allowedTypes> :
 
 def AnyRankedTensorOrMemRefType : RankedTensorOrMemRefOf<[AnyType]>;
 
+class RankedTensorOrScalarType<list<Type> allowedTypes> :
+  AnyTypeOf<[RankedTensorOf<allowedTypes>, AnyTypeOf<allowedTypes>]>;
+  
+def AnyRankedTensorOrScalarType :
+  RankedTensorOrScalarType<[AnySignlessIntegerOrIndex, AnyFloat]>;
+
+//===----------------------------------------------------------------------===//
+// Enum definitions
+//===----------------------------------------------------------------------===//
+
+def IREELinalgExt_IteratorType : I32EnumAttr<"IteratorType", "Iterator type", [
```

Comment:
Is it possible to reuse the iterator types from linalg? Or is this to avoid this dependency?

Reply:
+1, the iterator type enum attr is not reusable last I tried, but the `utils::IteratorType` enum itself should be reusable similar to here: https://github.com/iree-org/iree/blob/fa44a32bf05036e9ce707317f0650ae165b74a94/compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.td#L22

Reply:
Changes to reuse iterator types, but seems like a bit of a footgun. If we ever have enums in LinalgExt dialect we will end up with duplicate definitions of the `utils::IteratorType` enum.